Fixes #68201 - adds qs.* label formatting tokens

eamodio opened the pull request:
Addresses https://github.com/Microsoft/vscode/issues/68201

Allows `${qs.*}` tokens in the `resourceLabelFormatters.formatting.label`, where the `*` is resolved to a property of the querystring json object (assuming it is a json object)

/cc @isidorn

isidorn commented:
Thanks for your PR.
However this milestone we are not focused on adding new features (more about that in the iteratino plan).
Thus putting this to on-deck for now.

eamodio commented:
Doh -- oh sir, it's only wafer thin 🤣

eamodio commented:
@isidorn Any chance for this soon?

isidorn commented:
@eamodio you are the only user that wants this currently. I understand we could use this to better provide label in the explorer for the GitLens or RemoteHub extension but could you point me to a feature request for those extensions such that I possible get motivated to merge this in.

eamodio commented:
@isidorn I don't have any feature request that cover this (mainly because my uri hacks mostly provide a decent experience). It is more of debt that extensions have to pay in formatting uris in strange ways to get them to display nicely in the UI.

isidorn commented:
Ok, that is good enough. This is nice work, so I would merge it in if we add some tests.
Here's a code pointer for tests https://github.com/Microsoft/vscode/blob/73643c60a644145c6a57b61787efd6a62e153cfa/src/vs/workbench/services/label/test/label.test.ts#L22

eamodio commented:
@isidorn sorry for the long delay on this. I've rebased it with `master` and added a handful of tests. Let me know what you think. 🤞 this can still make it for July.

eamodio commented:
@isidorn any chance? 😉

isidorn commented:
@eamodio sorry for the slow response, I was on a 3 week vacation.
Thanks for adding tests. Merging in
